app/models/ext_management_system.rb

```diff
@@ -50,6 +50,18 @@ def self.supported_for_catalog?
     catalog_types.present?
   end
 
+  def self.label_mapping_classes
+    supported_subclasses.select(&:supports_label_mapping?)
+  end
+
+  def self.label_mapping_prefixes
+    label_mapping_classes.map(&:label_mapping_prefix).uniq
+  end
+
+  def self.entities_for_label_mapping
+    label_mapping_classes.reduce({}) { |all_mappings, klass| all_mappings.merge(klass.entities_for_label_mapping) }
+  end
```

Comment:
Please mark any of the above as private if they are not to be used as public API of the class.

Reply:
Can you also add specs for these methods?

Reply:
I added specs but we need public API, at least for now for UI.